bee-network/src/network/host.rs

```diff
@@ -0,0 +1,252 @@
+
+                    tokio::select! {
+                        _ = &mut shutdown => break,
+                        event = swarm_next_event => {
+                            process_swarm_event(event, &internal_event_sender, &peerlist).await;
+                        }
+                        command = recv_command => {
+                            if let Some(command) = command {
+                                process_command(command, &mut swarm, &peerlist).await;
+                            }
+                        },
+                    }
+                }
+
+                info!("Network Host stopped.");
+            });
+
+            info!("Network Host started.");
+
+            Ok(Self::default())
+        }
+    }
+}
+
+pub mod standalone {
+    use super::*;
+
+    use futures::channel::oneshot;
+
+    pub struct NetworkHost {
+        pub shutdown: oneshot::Receiver<()>,
+    }
+
+    impl NetworkHost {
+        pub fn new(shutdown: oneshot::Receiver<()>) -> Self {
+            Self { shutdown }
+        }
+
+        pub async fn start(self, config: NetworkHostConfig) {
+            let NetworkHost { mut shutdown } = self;
+            let NetworkHostConfig {
+                local_keys,
+                bind_multiaddr,
+                internal_event_sender,
+                mut internal_command_receiver,
+                peerlist,
+            } = config;
+
+            let mut swarm = start_swarm(local_keys.clone(), bind_multiaddr, internal_event_sender.clone()).await;
+
+            tokio::spawn(async move {
+                loop {
+                    let swarm_next_event = Swarm::next_event(&mut swarm);
+                    let recv_command = (&mut internal_command_receiver).recv();
+
+                    tokio::select! {
+                        _ = &mut shutdown => break,
+                        event = swarm_next_event => {
+                            process_swarm_event(event, &internal_event_sender, &peerlist).await;
+                        }
+                        command = recv_command => {
+                            if let Some(command) = command {
+                                process_command(command, &mut swarm, &peerlist).await;
+                            }
+                        },
+                    }
+                }
+
+                info!("Network Host stopped.");
+            });
+
+            info!("Network Host started.");
+        }
+    }
+}
+
+async fn start_swarm(
+    local_keys: Keypair,
+    bind_multiaddr: Multiaddr,
+    internal_event_sender: InternalEventSender,
+) -> Swarm<SwarmBehavior> {
+    let mut swarm = build_swarm(&local_keys, internal_event_sender)
+        .await
+        .expect("Fatal error: creating transport layer failed.");
+
+    info!("Binding to: {}", bind_multiaddr);
+
+    let _ = Swarm::listen_on(&mut swarm, bind_multiaddr).expect("Fatal error: binding address failed.");
```

Comment:
no, we can ignore the return type. But I changed it even to `let _listener_id = ` because I still find it nice if code documents itself. I might use it at some point. Better than purely ignoring it like a did above.